vue/sbc-common-components/src/components/SbcFeeSummary.vue

```diff
@@ -1,163 +1,152 @@
-                  entityType : for example CP
-
-               */
-  props: {
-    payURL: String,
-    filingData: {
-      type: Array,
-      default: function () {
-        return [{
-          filingDescription: '',
-          filingTypeCode: '',
-          entityType: ''
-        }]
-      }
+             the props accept an array of filing data item's
+                             the fee should have below items
+                             filingDescription : not mandatory.If exists , it will be displayed. Or else what ever from service call will be shown
+                             filingTypeCode : mandatory.Feecodes like OTADD ,OTANN
+                             entityType : for example CP
+
+ */
+    @Prop()
+    filingData: { filingDescription: string, filingTypeCode: string, entityType: string }[]
+
+    @Prop(String)
+    payURL: string
+
+    fees: Fee[] = []
+    fetchError: string = ''
+
+    mounted () {
+      // console.log('%c FeeMdoule-Data Recieved on Mount as %s %s', 'color: blue ;font-size : 12px', JSON.stringify(this.filingData), this.payURL)
+      FeeServices.getFee(this.filingData, this.payURL).then(data => {
+        this.fetchError = ''
+        this.fees = data
+        this.$emit('total-fee', this.totalFilingFees)
+      }).catch((error: any) => {
+        this.fetchError = 'Error fetching fees' + error
+      })
     }
-  },
-
-  data: () => ({
-    fees: [],
-    fetchError: ''
-  }),
-
-  mounted () {
-    // console.log('%c FeeMdoule-Data Recieved on Mount as %s %s', 'color: blue ;font-size : 12px', JSON.stringify(this.filingData), this.payURL)
-    FeeServices.getFee(this.filingData, this.payURL).then(data => {
-      this.fetchError = ''
-      this.fees = data
-    }).catch((error: any) => {
-      this.fetchError = 'Error fetching fees' + error
-    })
-  },
-  computed: {
-    totalFilingFees () :number {
-      let totalFee = 0
-      if (this.fees) {
-        // console.log('%c FeeMdoule-Watch ZERO FEE %s', 'color: blue ;font-size : 12px', JSON.stringify(this.filingData))
-        totalFee = this.fees.reduce((acc: number, item: { fee: number; }) => acc + item.fee, 0)
-      }
-      this.$emit('total-fee', totalFee)
-      return totalFee
+
+    get totalFilingFees () {
+      return this.fees.reduce((acc: number, item: { fee: number; }) => acc + item.fee, 0)
     }
-  },
-  watch: {
-    filingData: function (newVal) {
-      // console.log('%c FeeMdoule-Watch Activated as %s', 'color: blue ;font-size : 12px', JSON.stringify(this.filingData))
+
+    @Watch('filingData')
+    onFilingDataChanged (val: string, oldVal: string) {
+      console.log('%c FeeMdoule-Watch Activated as %s', 'color: blue ;font-size : 12px', JSON.stringify(this.filingData))
       FeeServices.getFee(this.filingData, this.payURL).then((data: any) => {
         this.fetchError = ''
         this.fees = data
+        this.$emit('total-fee', this.totalFilingFees)
```

Comment:
There is a more standard TS way to emit an event. Have a look at BaseAddress.vue.

Reply:
Fixed in next commit.